python_modules/dagster/dagster/_config/structured_config.py

```diff
@@ -209,20 +210,32 @@ def _convert_pydantic_field(pydantic_field: ModelField) -> Field:
             pydantic_field.type_, description=pydantic_field.field_info.description
         )
 
-    dagster_type = pydantic_field.type_
     if pydantic_field.shape != SHAPE_SINGLETON:
         raise NotImplementedError(f"Pydantic shape {pydantic_field.shape} not supported")
 
-    inner_config_type = convert_potential_field(dagster_type).config_type
     return Field(
-        config=inner_config_type,
+        config=_config_type_for_pydantic_field(pydantic_field),
         description=pydantic_field.field_info.description,
         default_value=pydantic_field.default
         if pydantic_field.default
         else FIELD_NO_DEFAULT_PROVIDED,
     )
 
 
+def _config_type_for_pydantic_field(pydantic_field: ModelField):
```

Comment:
I think we will also want to invoke this [here](https://github.com/dagster-io/dagster/pull/11386/files#diff-e552cc00c002d5aca0c5051d6024b3a5e0ae02d520fef8086d017c36735044faR267), to handle cases without a Config class such as:

```python
@op
def my_op(config: str):
    pass
```